lib/db_connection/connection.ex

```diff
@@ -169,23 +169,25 @@ defmodule DBConnection.Connection do
         mon = Process.monitor(pid)
         handle_checkout({ref, mon}, timeout, from, state, s)
       %{client: :closed} ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of disconnection"
+        err = DBConnection.Error.exception(message)
         {:reply, {:error, err}, s}
       %{queue: queue} when queue? == true ->
         client = {ref, Process.monitor(pid)}
         queue = :queue.in({client, timeout, from}, queue)
         {:noreply, %{s | queue: queue}}
       _ when queue? == false ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of queue"
```

Comment:
A good error message needs to have enough information so developers can make a decision of if and how they should act on it. "because of queue" does not give me any information to say if it is a configuration error, a temporary error or something I should investigate. So maybe I would say "connection not available and queueing is disabled".